thrust/thrust/system/cuda/detail/swap_ranges.h

```diff
@@ -37,66 +37,44 @@
 #endif // no system header
 
 #if _CCCL_HAS_CUDA_COMPILER
-#  include <thrust/distance.h>
-#  include <thrust/swap.h>
-#  include <thrust/system/cuda/detail/par_to_seq.h>
-#  include <thrust/system/cuda/detail/parallel_for.h>
-#  include <thrust/system/cuda/detail/transform.h>
 
-#  include <cuda/std/utility>
+#  include <thrust/iterator/iterator_traits.h>
+#  include <thrust/system/cuda/detail/parallel_for.h>
 
-#  include <iterator>
+#  include <cuda/std/__algorithm_>
+#  include <cuda/std/iterator>
 
 THRUST_NAMESPACE_BEGIN
 
 namespace cuda_cub
 {
-
-namespace __swap_ranges
-{
-
 template <class ItemsIt1, class ItemsIt2>
-struct swap_f
+struct __swap_f
 {
   ItemsIt1 items1;
   ItemsIt2 items2;
 
-  using value1_type = thrust::detail::it_value_t<ItemsIt1>;
-  using value2_type = thrust::detail::it_value_t<ItemsIt2>;
-
-  THRUST_FUNCTION
-  swap_f(ItemsIt1 items1_, ItemsIt2 items2_)
-      : items1(items1_)
-      , items2(items2_)
-  {}
-
   template <class Size>
-  void THRUST_DEVICE_FUNCTION operator()(Size idx)
+  _CCCL_HOST_DEVICE void operator()(Size idx) const
   {
     // TODO(bgruber): this should probably use ::cuda::std::iter_swap(items1 + idx, items2 + idx);
-    value1_type item1 = items1[idx];
-    value2_type item2 = items2[idx];
+    thrust::detail::it_value_t<ItemsIt1> item1 = items1[idx];
+    thrust::detail::it_value_t<ItemsIt2> item2 = items2[idx];
     using ::cuda::std::swap;
     swap(item1, item2);
     items1[idx] = item1;
     items2[idx] = item2;
```

Comment:
Do we want to directly use iter_swap?

Reply:
We can, but it required to fix `swap` for `device_reference`.